### PR TITLE
[Helm Chart] Make use `compression` and GA rotated logs

### LINKED ISSUE #11559: [Helm chart] Adjust Elastic Agent Helm Chart as ingesting GZIP file is GA on 9.3.0
Refactor the Elastic Agent Helm Chart account for ingesting GZIP files being GA and on by default on filebeat 9.3.0.

### Acceptance Criteria

- Remove `gzip_experimental` from template.
- keep rotated log ingestion as opt-in
- If gzip_experimental is passed by a user in values.yaml during an upgrade, render it harmlessly or log a deprecation warning (via NOTES.txt?).

### Corner Cases

Test a Helm upgrade from the previous version to 9.2.0 with custom values to ensure the release does not fail due to unknown config keys.

### PR DESCRIPTION
**Pull request by AndersonQ**
## What does this PR do?

It replaces `gzip_experimental` by `compression` and removes the BETA status of the agent's helm-chart container logs

## Why is it important?

Ingesting GZIP-compressed logs is GA on filebeat 9.3.0. It deprecates `gzip_experimental` and replaces it by `compression`.
## Checklist

- [x] I have read and understood the [pull request guidelines](https://github.com/elastic/elastic-agent/blob/main/CONTRIBUTING.md#pull-request-guidelines) of this project.
- [x] My code follows the style guidelines of this project
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- ~~[ ] I have added an integration test or an E2E test~~ the existing tests are enough.  

## Disruptive User Impact

None

## Notes for the reviewer

There is a bit of a chicken-egg problem here as this PR requires https://github.com/elastic/beats/pull/47893 to be merged and a snapshot to be published. And both need to go into 9.3.
~So ideally we merge it just before the FF. Or I can keep both flags either after the FF or on a next release :/~

I believe the best is wait close to the FF to merge it, if we need, we can skip the test while the beats snapshot isn't available. It'll be the least disruptive for the users. 

## How to test this PR locally

- build the docker image with an agentbeat/filebeat containing the changes of https://github.com/elastic/beats/pull/47893:
```shell
rm -rf ../beats/build ../beats/x-pack/agentbeat/build ../beats/x-pack/filebeat/build
DEV=true SNAPSHOT=true EXTERNAL=false DOCKER_VARIANTS="basic" PACKAGES="docker" PLATFORMS=linux/amd64 mage -v package
```
 - load it into docker
```shell
docker load -i ./build/distributions/docker.elastic.co/elastic-agent/elastic-agent:9.3.0-SNAPSHOT
```

- start a elastic stack
- create a kind cluster
```shell
kind create cluster --name v1.33.0-kubernetes
```
- load the necessary images into kind
```shell
kind load docker-image --name v1.33.0-kubernetes docker.elastic.co/elastic-agent/elastic-agent:9.3.0-SNAPSHOT docker.elastic.co/elastic-agent/elastic-agent:9.2.2 registry.k8s.io/kube-state-metrics/kube-state-metrics:v2.16.0
```

- deploy the agent using the current helm chart
    - on your deployment, go to `app/observabilityOnboarding/kubernetes/?category=kubernetes` and follow the steps. It'll be something like:
```shell
helm repo add elastic https://helm.elastic.co/ && \
helm install elastic-agent elastic/elastic-agent \
  --version 9.2.2 \
  -n kube-system \
  --set outputs.default.url=https:\/\/YOUR_DEPLOYMENT.elastic-cloud.com:443 \
  --set kubernetes.onboardingID=YOUR_ONBORDING_ID \
  --set kubernetes.enabled=true \
  --set outputs.default.type=ESPlainAuthAPI \
  --set outputs.default.api_key=$(echo "YOUR_API_KEY" | base64 -d)
```
- verify all is working, container logs are being ingested
- deploy a flog container

<details>
<summary>
flog.yaml:
</summary>

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: flog-log-generator-deployment
  labels:
    app: flog-log-generator
spec:
  replicas: 1
  selector:
    matchLabels:
      app: flog-log-generator
  template:
    metadata:
      labels:
        app: flog-log-generator
    spec:
      containers:
        - name: flog-unstructured-cont-rot
          image: mingrammer/flog
          args: ["-t", "stdout", "-d", "1us", "-l"]
      restartPolicy: Always
```
</details>

```shell
kubectl apply -f ./flog.yaml
```

- check the logs are being ingested. Use the filter `log.file.path : *flog*`. Logs are ingested from `/var/log/containers/`
- package the helm chart
```shell
mage helm:package
```
- update the helm chart to collect rotated logs:
```shell
helm upgrade elastic-agent elastic-agent-9.3.0-beta.tgz \
  -n kube-system \
  --set kubernetes.containers.logs.rotated_logs=true \
  --set outputs.default.url=https:\/\/YOUR_DEPLOYMENT.elastic-cloud.com:443 \
  --set kubernetes.onboardingID=YOUR_ONBORDING_ID \
  --set kubernetes.enabled=true \
  --set outputs.default.type=ESPlainAuthAPI \
  --set outputs.default.api_key=$(echo "YOUR_API_KEY" | base64 -d)
```
- check the rotated logs, plain and gzip, were ingested. Use the following filters `log.file.path : *flog*log.*` and `log.file.path : *flog*log.*.gz`:
- if you're not seeing gzip log, make sure filebeat will encounter unique gzip-compressed logs by:
  - removing the agent: `helm uninstall elastic-agent -n kube-system`
  - running flog for 30s or more to ensure logs are rotated at least twice: `kubectl delete -f /tmp/flog.yaml && kubectl apply -f /tmp/flog.yaml`
  - you might check the control-plane node logs folder:
    - `docker exec -it v1.33.0-kubernetes-control-plane bash`
    - `ls -l /var/log/pods/default<tab><tab>` -> `ls -l /var/log/pods/default_flog-log-generator-deployment-586ccfc5f5-s62jb_8db1912e-789d-44bc-92fa-fc5744a2913c/flog-unstructured-cont-rot/`
    - install the agent again. Use the same `helm upgrade`, but with install instead: `helm install elastic-agent elastic-agent-9.3.0-beta.tgz [...]`

## Related issues

- Closes #11559 
- Relates https://github.com/elastic/elastic-agent/pull/11129

## Questions to ask yourself

- How are we going to support this in production?
- How are we going to measure its adoption?
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

File: changelog/fragments/1765570441-helm-chart-container-logs-is-GA.yaml
Comment (AndersonQ) [inline changelog/fragments/1765570441-helm-chart-container-logs-is-GA.yaml:1]: As the last PR wasn't released, it does not make sense to have both entries

File: deploy/helm/elastic-agent/values.schema.json [hunk 1]
```diff
@@ -231,7 +231,7 @@
                                 },
                                 "rotated_logs": {
                                     "type": "boolean",
-                                    "description": "BETA: enable rotated logs ingestion. WARNING: enabling it might cause date re-ingestion on existing deployments. [ref](https://www.elastic.co/docs/reference/integrations/kubernetes/container-logs#ingesting-rotated-container-logs)."
+                                    "description": "enable rotated logs ingestion. WARNING: enabling it might cause date re-ingestion on existing deployments. [ref](https://www.elastic.co/docs/reference/integrations/kubernetes/container-logs#ingesting-rotated-container-logs)."
```
Comment (ycombinator) [inline deploy/helm/elastic-agent/values.schema.json]:
```suggestion
                                    "description": "enable rotated logs ingestion. WARNING: enabling it might cause data re-ingestion on existing deployments. [ref](https://www.elastic.co/docs/reference/integrations/kubernetes/container-logs#ingesting-rotated-container-logs)."
```

File: deploy/helm/elastic-agent/values.yaml [hunk 2]
```diff
@@ -151,7 +151,7 @@ kubernetes:
       # -- enable containers logs stream [ref](https://www.elastic.co/docs/current/integrations/kubernetes/container-logs)
       # @section -- 2 - Kubernetes integration
       enabled: true
-      # -- BETA: enable rotated logs ingestion. WARNING: enabling it might cause date re-ingestion on existing deployments. [ref](https://www.elastic.co/docs/reference/integrations/kubernetes/container-logs#ingesting-rotated-container-logs)
+      # -- enable rotated logs ingestion. WARNING: enabling it might cause date re-ingestion on existing deployments. [ref](https://www.elastic.co/docs/reference/integrations/kubernetes/container-logs#ingesting-rotated-container-logs)
```
Comment (ycombinator) [inline deploy/helm/elastic-agent/values.yaml]:
```suggestion
      # -- enable rotated logs ingestion. WARNING: enabling it might cause data re-ingestion on existing deployments. [ref](https://www.elastic.co/docs/reference/integrations/kubernetes/container-logs#ingesting-rotated-container-logs)
```

**Comment by AndersonQ** [comment]
@ycombinator I fixed the typos. Thanks for spotting it!

regarding the failing test, It's failing because the filebeat PR isn't merged yetr.
So, I think the best it to skip the test while the PR on filebeat isn't merged.

Therefore, my proposal is to wait until tomorrow, if the PR isn't merged, I skip the test on this PR and as soon as the filebeat PR is merged I remove the test skip. Do you agree?

It's better than keeping the old flag which will be logging an unnecessary warning to the users.

**Comment by AndersonQ** [comment]
> @ycombinator I fixed the typos. Thanks for spotting it!
> 
> regarding the failing test, It's failing because the filebeat PR isn't merged yetr. So, I think the best it to skip the test while the PR on filebeat isn't merged.
> 
> Therefore, my proposal is to wait until tomorrow, if the PR isn't merged, I skip the test on this PR and as soon as the filebeat PR is merged I remove the test skip. Do you agree?
> 
> It's better than keeping the old flag which will be logging an unnecessary warning to the users.

I create a PR to already skip the test as it'll fail anyway once the beats PR make to the snapshot. So once https://github.com/elastic/elastic-agent/pull/11815 is merged, I'll update the branch and un-skip the test :)

**Comment by AndersonQ** [comment]
I had to re-add the test skip as even though https://github.com/elastic/beats/pull/47893 is merged and made to a snapshot, it did not make to the one the test are using. 

I was investigating and I drill down to the following:
 - snapshot used: 9.3.0-43b60b19
 - downloads available at https://snapshots.elastic.co/9.3.0-43b60b19/manifest-9.3.0-SNAPSHOT.json
 - beats version:
```
./agentbeat filebeat version
filebeat version 9.3.0 (amd64), libbeat 9.3.0 [72314a046f861de2e6b1bc3ae1beb8d979e72cd2 built 2025-12-12 23:46:58 +0000 UTC] (FIPS-distribution: false)
```
 - commit: https://github.com/elastic/beats/commits/72314a046f861de2e6b1bc3ae1beb8d979e72cd2
 - last commit from 13th Dec

if you run the build locally and package the right version of filebeat/agentbeat all works

**Comment by AndersonQ** [comment]
> LGTM. Please put up a PR to unskip the test so we don't forget about it. I suspect it'll go green in the next day or two. Thanks.

done:
 - https://github.com/elastic/elastic-agent/pull/11857
